### ISSUE TITLE
Add something like `docker rm --remove-all-stopped`

### ISSUE THREAD

**Issue by JonasT** [docker/docker#17868]
Please add something like `docker rm --remove-all-stopped`. (or a syntactic variation that is similarly simple and, most importantly, documented on a future `docker rm --help` page.)

You can add all sorts of red warnings to prevent those who actually want to keep their stopped containers for data storage or whatever from accidentally deleting them, but the fact is that for many people's setups, this command is needed.

---
**Why docker rm $(docker ps -aq) is NOT a good replacement:**
Many of you are quick to suggest `docker rm $(docker ps -aq)` under the misconception that it is an acceptable replacement. Please note, the simple fact that each of you are so quick to suggest this somewhat shows that it is a commonly asked command. However, what some seem to forget is the fact that *this shell command is nothing which a non-professional docker user could easily come up with on their own.* Yes, it can be googled and copied from various pages, but this suggestion is simply a bad replacement/workaround for what should have been an easy-to-use integrated command to start with. It's just a bad user experience.

**Comment by cpuguy83** [comment]
-1
The shell replacement is not bad, it is designed that way. All the CLI options are.

**Comment by phemmer** [comment]
Sounds like the entire argument would be resolved by documenting the command.

Perhaps it's time to create this "tips and tricks" document people keep saying we should create.

**Comment by JonasT** [comment]
IMHO whatever it is, it should be in `docker rm --help`. That is where people are going to check first when they are trying to get rid of all their old garbage.

Having it somewhere in the docs too is great, but it's not a solution for the current problem that people will need to *search* for this information instead of having it easily in their face when checking `--help` which is how it should be. Even if this information is added the docs, you don't want people to be *required* to dig through the large pile of documentation to find this. This would be basically just sticking to the "well let people just search it" way that it is right now.

Considering this, I think a new switch is really the best choice. If you need to place some shell command scripting in your `--help` section to use your tool, it's probably time to improve the available options.

**Comment by phemmer** [comment]
One of the core unix philosophies is relevant here: http://www.catb.org/~esr/writings/taoup/html/ch01s06.html#id2877684

> **Rule of Composition: Design programs to be connected with other programs.**
>
> Unix tradition strongly encourages writing programs that read and write simple, textual, stream-oriented, device-independent formats. Under classic Unix, as many programs as possible are written as simple filters, which take a simple text stream on input and process it into another simple text stream on output.
>
> Despite popular mythology, this practice is favored not because Unix programmers hate graphical user interfaces. It's because if you don't write programs that accept and emit simple text streams, it's much more difficult to hook the programs together

Gluing programs together is a very common practice in unix land.
Lets pretend we added a `--remove-stopped-containers`. Next someone is going to want to remove stopped containers which exited with status != 0. Then containers which are more than a week old. Then containers were only run interactively.
By instead providing a flexible method with which users can craft any filter they want, we let users accomplish all those things, and much more, while avoiding the feature bloat.

**Comment by yoanisgil** [comment]
@phemmer creating it it's only half of the problem (personally I think information is as good as the ability to find it). I think the point here is to be able to easily find this sort of documentation for non experts. However, I think I have to agree with @cpuguy83 but on the basis that if we add an option for removing all stopped containers, the same logic could apply to other uses cases like removing all running containers with docker rm -f $(docker ps -q), or any possible scenario someone could come up with (for instance clean images with docker rmi, etc)

@JonasT I think the tools are there and it's only a matter of figuring out how to use them. I think each docker command does it's job pretty well and you can rely on Unix-like utilities like shell expansion, pipes, etc to create new functionalities. But that's just my humble opinion.

**Comment by yoanisgil** [comment]
@phemmer you just stole my comment :D. Anyway +1

**Comment by JonasT** [comment]
`docker rm` isn't obviously suitable for shell use, because it's not obvious without -f it won't remove non-stopped containers and the usual formatted table output makes it more seem like an interactive thing.

That doesn't mean you can't recommend this as an official way (with the shell use instead of an explicit option). However, no matter what you do, I still think this basic operation really should be described in `--help` with the full command necessary for this. And while encouraging the Unix philosophy of combining many tools is good, to me docker rm simply doesn't seem to be designed in a way that makes it obviously suitable for that kind of use without some research *which shouldn't be necessary for this operation (IMHO)*.

---
TL;DR: I don't care that much about the details, but please do more than just adding this somewhere in the docs. It's too much hidden away right now, the tool itself should afford itself more obviously to this use with some obvious hint or description (or a new switch just for this operation) in `--help`.

**Comment by JonasT** [comment]
Just to make this more clear:

>  Lets pretend we added a --remove-stopped-containers. Next someone is going to want to remove stopped containers which exited with status != 0. Then containers which are more than a week old. Then containers were only run interactively.

Yes, but I still think this operation specifically is more important than those special cases. Again, I have seen all people using docker in my close environment search **this exact command** on the web because it's not obvious how to do it.

**Comment by calavera** [comment]
I agree with Brian. We're not going to add this flag. Thanks for your suggestion but I'm going to close it as "won't fix".
